### PR TITLE
Left-Right Motors on Shooter.java

### PR DESCRIPTION
**Pull request by Ash1R**
Converted top down to left right, also added some constants for RPMs for spin.

### CODE REVIEW

File: src/main/java/frc/robot/constants/ShooterConstants.java [hunk 1]
```diff
@@ -3,8 +3,14 @@
...
     // motor Ids
-    public static final int BOTTOM_MOTOR_ID = 1;
-    public static final int TOP_MOTOR_ID = 2;
+    public static final int RIGHT_MOTOR_ID = 1;
+    public static final int LEFT_MOTOR_ID = 2;
+
+    // RPMs
+
+    public static final double LEFT_SHOOT_RPM = 6000;
+    public static final double RIGHT_SHOOT_RPM = 4000;
```
Comment (Kyle-Eldridge) [inline src/main/java/frc/robot/constants/ShooterConstants.java:12]: Was our shoot redesigned to have spin, or is there another reason you're running the motors at different speeds?

File: src/main/java/frc/robot/subsystems/gpm/Shooter.java [hunk 2]
```diff
@@ -43,239 +44,244 @@ public class Shooter extends SubsystemBase {
...
-
-	// bottom motor
-    private final CANSparkFlex bottomMotor = new CANSparkFlex(ShooterConstants.BOTTOM_MOTOR_ID, MotorType.kBrushless);
-    private final RelativeEncoder bottomMotorEncoder = bottomMotor.getEncoder();
+	private final PIDController leftPID = new PIDController(P, I, D);
+	private FlywheelSim leftFlywheelSim;
+	private double leftMotorSpeedSim;
+	private double leftPower = 0.0;
+
+	// bottrightor
```
Comment (Kyle-Eldridge) [inline src/main/java/frc/robot/subsystems/gpm/Shooter.java:55]: typo

File: src/main/java/frc/robot/subsystems/gpm/Shooter.java [hunk 3]
```diff
@@ -43,239 +44,244 @@ public class Shooter extends SubsystemBase {
...
+	// TODO: TUNE THIS
+	private final SimpleMotorFeedforward feedforward = new SimpleMotorFeedforward(S, V);
+
+	public Shooter() {
 		// set the RPM tolerance of the PID controllers
-        topPID.setTolerance(TOLERANCE);
-        bottomPID.setTolerance(TOLERANCE);
+		leftPID.setTolerance(TOLERANCE);
+		rightPID.setTolerance(TOLERANCE);
 		// invert the bottom motor so +power sends the note out
```
Comment (Kyle-Eldridge) [inline src/main/java/frc/robot/subsystems/gpm/Shooter.java:71]: Change this comment and any other references to top/bottom motors.

File: src/main/java/frc/robot/subsystems/gpm/Shooter.java [hunk 4]
```diff
@@ -43,239 +44,244 @@ public class Shooter extends SubsystemBase {
...
+		SmartDashboard.putBoolean("at setpoint?", atSetpoint());
+	}
 
 	@Override
 	public void simulationPeriodic() {
 		// assume the battery voltage is 12 volts
 		double voltage = 12.0;
 
-		// topPower and bottomPower are now member variables; cannot read them from simulated encoder
+		// topPower and bottomPower are now member variables; cannot read them from
```
Comment (Kyle-Eldridge) [inline src/main/java/frc/robot/subsystems/gpm/Shooter.java:107]: Again, change comment. Also remove commented out code.

File: src/main/java/frc/robot/subsystems/gpm/Shooter.java [hunk 5]
```diff
@@ -43,239 +44,244 @@ public class Shooter extends SubsystemBase {
...
+	 * Gets the difference in RPM between the top motor and bottom motor.
+	 * 
+	 * @return the top motor's RPM - the bottom motor-s RPM
+	 * @see getMotorSpeedDifference
+	 * @see getTopMotorRPM
+	 * @see getBottomMotorRPM
+	 * @see atSetpoint
+	 */
 	public double getMotorRPMDifference() {
 		return getTopMotorRPM() - getBottomMotorRPM();
```
Comment (Kyle-Eldridge) [inline src/main/java/frc/robot/subsystems/gpm/Shooter.java:272]: These methods still reference top/bottom motors.

**Review by Kyle-Eldridge — APPROVED** [review]
A few very small changes, mostly good.

File: src/main/java/frc/robot/constants/ShooterConstants.java (hunk 1 above)
Comment (Ash1R) [reply src/main/java/frc/robot/constants/ShooterConstants.java:12]: Yes, we have spin now. I used RPMs from Mechanical Advantage's shooter which is mechanically similar to ours, but we're going to end up fine tuning these values anyways.

File: src/main/java/frc/robot/subsystems/gpm/Shooter.java (hunk 2 above)
Comment (Ash1R) [reply src/main/java/frc/robot/subsystems/gpm/Shooter.java]: yer

File: src/main/java/frc/robot/subsystems/gpm/Shooter.java (hunk 3 above)
Comment (Ash1R) [reply src/main/java/frc/robot/subsystems/gpm/Shooter.java]: yer

File: src/main/java/frc/robot/subsystems/gpm/Shooter.java (hunk 5 above)
Comment (Ash1R) [reply src/main/java/frc/robot/subsystems/gpm/Shooter.java]: not anymore

File: src/main/java/frc/robot/subsystems/gpm/Shooter.java (hunk 4 above)
Comment (Ash1R) [reply src/main/java/frc/robot/subsystems/gpm/Shooter.java]: commented out code is so aesthetic though!
(it's removed)

**Comment by Ash1R** [comment]
@chem7397 could u review this

**Review by MichaelChekler — APPROVED** [review]

**Pull request merged**